File: papi/lista1/exercicio16.py
while True:
    try:
        lado1 = float(input("Por favor digite o comprimento do primeiro lado do triângulo: "))
        break
    except ValueError:
        print("Por favor digite um número válido, usando o ponto como separador decimal")

while True:
    try:
        lado2 = float(input("Por favor digite o comprimento do segundo lado do triângulo: "))
        break
    except ValueError:
        print("Por favor digite um número válido, usando o ponto como separador decimal")

while True:
    try:
        lado3 = float(input("Por favor digite o comprimento do terceiro lado do triângulo: "))
        break
    except ValueError:
        print("Por favor digite um número válido, usando o ponto como separador decimal")

if lado1 == lado2 and lado1 == lado3:
    print("O triângulo é equilátero")
elif lado1 == lado2 or lado1 == lado3 or lado2 == lado3:
    print("O triângulo é isósceles")
else:
    print("O triângulo é escaleno")

File: papi/lista1/exercicio16.kt
package lista1

fun main() {
    println("Vamos determinar o tipo de um triângulo pelos seus lados.")
    var lados: ArrayList<Double> = ArrayList()
    var cont = 1
    while (cont < 4) {
        while (true) {
            try {
                print("Digite o valor do lado $cont: ")
                var input = readLine()
                lados.add(input!!.toDouble())
                if (lados[cont - 1] < 0) {
                    // Vamos lançar uma exception, mas temos que remover o valor do ArrayList
                    // lados, pois já o inserimos
                    lados.removeAt(cont - 1)
                    throw NumberFormatException()
                }
                break
            } catch (e: NumberFormatException) {
                println("Valor inválido. Por favor, digite um valor numérico maior que zero.")
            }
        }
        cont++
    }

    if (lados[0] == lados[1] && lados[1] == lados[2]) {
        println("O triângulo é equilátero.")
    } else if (lados[0] == lados[1] || lados[1] == lados[2] || lados[0] == lados[2]) {
        println("O triângulo é isósceles.")
    }  else {
        println("O triângulo é escaleno.")
    }
}